vhdbuilder/publish/Marketplace/new-sku-and-add-image-version.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash -e
+
+required_env_vars=(
+    "SKU_PREFIX"
+    "SKU_TEMPLATE_FILE"
+    "AZURE_TENANT_ID"
+    "AZURE_CLIENT_ID"
+    "AZURE_CLIENT_SECRET"
+    "PUBLISHER"
+    "OFFER"
+    "CONTAINER_RUNTIME"
+)
+
+(set -x; ls -lhR artifacts )
+
+for v in "${required_env_vars[@]}"
+do
+    if [ -z "${!v}" ]; then
+        echo "$v was not set!"
+        exit 1
+    fi
+done
+
+if [ ! -f "$SKU_TEMPLATE_FILE" ]; then
+    echo "Could not find sku template file: ${SKU_TEMPLATE_FILE}!"
+    exit 1
+fi
+
+(set -x; ls -lhR artifacts )
+vhd_artifacts_path="publishing-info-2019"
+if [[ ${CONTAINER_RUNTIME} = "containerd" ]]; then
+    vhd_artifacts_path="publishing-info-2019-containerd"
+fi
+VHD_INFO="artifacts/vhd/${vhd_artifacts_path}/vhd-publishing-info.json"
+if [ ! -f "$VHD_INFO" ]; then
+    echo "Could not find VHD info file: ${VHD_INFO}!"
+    exit 1
+fi
+
+short_date=$(date +"%y%m")
+pretty_date=$(date +"%b %Y")
+sku_id="${SKU_PREFIX}-${short_date}"
+echo "Checking if offer contains SKU: $sku_id"
+# Check if SKU already exists in offer
+(set -x; hack/tools/bin/pub skus list -p $PUBLISHER -o $OFFER | jq ".[] | .planId" | tr -d '"' | tee skus.txt)
+echo ""
+if grep -q $sku_id skus.txt; then
+    echo "Offer already has SKU"
+else
+    echo "Creating new SKU"
+    < $SKU_TEMPLATE_FILE sed s/{{ID}}/"$sku_id"/ | sed s/{{MONTH-YEAR}}/"$pretty_date/" | sed s/{{CONTAINER_RUNTIME}}/"$CONTAINER_RUNTIME/" > sku.json
+    echo "" ; cat sku.json ; echo ""
+    (set -x ; hack/tools/bin/pub skus put -p $PUBLISHER -o $OFFER -f sku.json ; echo "")
```

Comment:
What will happen if publishing sku fails?

Reply:
Changed it to exit on any error.

Reply:
What do you mean for the change, removing "set -x"?

Reply:
sorry, forgot to mention this. I set -xue` on top of this script https://github.com/Azure/AgentBaker/pull/559/files#diff-1cce0e10b1876580cd1745719db441a2a39087dead697c03d45c2cc9e7ec1db0R1

Reply:
@mainred adding a comment should be better.